src/index.ts

```diff
@@ -16,27 +16,23 @@ export class CircularBuffer<T> {
 	}
 
 	get size(): number {
-		return this.#size
+		return this.#overflow ? this.#maxSize : this.#next
 	}
 
 	push(value: T): void {
 		this.#data[this.#next] = value
-
-		if (this.#size !== this.#maxSize) {
-			this.#size += 1
-		}
-
 		if (this.#next === this.#maxSize - 1) {
 			this.#next = 0
+			this.#overflow = true
 		} else {
 			this.#next += 1
 		}
 	}
 
 	toArray(): Array<T> {
-		if (this.#size < this.#maxSize) {
+		if (this.size < this.#maxSize) {
```

Comment:
We can just check `!this.#overflow` here

Reply:
Oh yeah you're right !